controllers/userController.js

```diff
@@ -0,0 +1,137 @@
+const jwt = require("jsonwebtoken");
+const createError = require("http-errors");
+const User = require("../models/User");
+
+const getUserInfo = async (req, res, next) => {
+  const { id } = req.params;
+
+  try {
+    const user = await User.findById(id).exec();
+
+    res.json({
+      result: user,
+    });
+  } catch (err) {
+    console.error(err);
+    next(err);
+  }
+};
+
+const getGuestBook = async (req, res, next) => {
+  const { id } = req.params;
+
+  try {
+    const user = await User.findById(id).exec();
+    const userGuestBook = user.guestBook;
+
+    res.json({ result: { guestBook: userGuestBook } });
+  } catch (err) {
+    console.error(err);
+    next(err);
+  }
+};
+
+const addMessage = async (req, res, next) => {
+  const { id } = req.params;
+  const { message } = req.body;
+  const refershToken = req.cookies.jwt;
+  const isoDateTime = new Date().toISOString();
+  let userEmail;
+
+  try {
+    jwt.verify(
+      refershToken,
+      process.env.ACCESS_TOKEN_SECRET,
+      (err, decoded) => {
+        if (err) {
+          return createError(403, "Forbidden");
+        }
+
+        userEmail = decoded.email;
+      }
+    );
+
+    const user = await User.findOne({ email: userEmail }).exec();
+    const userName = user.name;
+
+    const newMessage = await User.findByIdAndUpdate(
+      id,
+      {
+        $push: {
+          guestBook: { name: userName, message, date: isoDateTime },
+        },
+      },
+      { new: true }
+    );
+
+    res.json({ result: newMessage });
+  } catch (err) {
+    console.error(err);
+    next(err);
+  }
+};
+
+const toggleItem = async (req, res, next) => {
+  const { id, itemId } = req.params;
+  const { from, to } = req.body;
+
+  try {
+    const item = await User.findByIdAndUpdate(
+      id,
+      {
+        $pull: { [from]: itemId },
+      },
+      { new: true }
+    );
+
+    await User.findByIdAndUpdate(id, {
+      $push: {
+        [to]: {
+          purchasedBy: item.purchasedBy,
+          name: item.name,
+          location: [0, 0],
+        },
+      },
+    });
+
+    const user = User.findById(id);
+
+    res.json({
+      result: {
+        inbox: user.inItemBox,
+        outBox: user.outItemBox,
+        presentBox: user.presentBox,
+      },
+    });
```

Comment:
```suggestion
   const targetItem = [];
    const restItems = [];

    const user = await User.findByIdAndUpdate(id);
    user[from].forEach((item) => {
      if (item._id.toString() === itemId) {
        targetItem.push(item);
      } else {
        restItems.push(item);
      }
    });

    user[from] = restItems;
    user[to].push(...targetItem);
    const success = await user.save();

    res.json({
      inbox: success.inItemBox,
      outBox: success.outItemBox,
      presentBox: success.presentBox,
    });
```

현재 모델에서 findById로 도큐먼트를 반복해서 불러오고 있어요, 같은 도큐먼트의 값을 업데이트 하는 부분이라면 `save()`도 하나의 옵션이 될 것 같습니다.
https://mongoosejs.com/docs/documents.html#updating-using-save

그리고 findOne메소드는 도큐먼트 자체를 리턴하기 때문에 79line item에 담기는 값은 특정 아이템 도큐먼트가 아닌 유저의 전체 정보가 담긴 도큐먼트일거에요! 따라서 87-95의 코드에서 업데이트가 되고 있지 않는 것 같습니다. 
포스트맨같은 프로그램을 사용해서 테스트를 진행하면 백앤드 api 작성이 훨씬 수월해 질 것 같습니다.  

은별님이 사용하신 $pull, $push로 해결해 보려고 했는데, subSchema를 핸들링 하는 부분이 역시 어렵네요..

Reply:
오.. save()하는 방법도 있었군요!
제가 너무 findByIdAndUpdate 메서드를 사용해야 한다는 틀에 박혀있었던것 같습니다.
subSchema 핸들링은 저도 아직 더 찾아봐야 할것 같아요..
지원님께서 작성해주신 코드 참고해서 수정하도록 하겠습니다!

이번 브랜치 작업을 하면서 지원님의 추천으로 postman으로 테스트를 처음 해보게 되었는데 바로바로 결과 확인이 가능하네요. 정말 유용하고 백엔드 작업의 경우에는 무조건적으로 사용하는게 좋을것 같네요. 조언 감사합니다!
찾아보니 postman에 workspace 를 만들어 팀끼리 테스트 공유가 가능하니 제가 초대해 드리도록 하겠습니다!

Reply:
오 팀공유는 저도 잘 몰랐었는데 테스트 하는 리퀘스트 서로 공유하면 좋을 것 같네요 :)
감사합니다~!